src/Raven.Server/Documents/Indexes/Persistence/Corax/OutputReduceCoraxIndexWriteOperation.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Indexes;
+using Raven.Server.Documents.Indexes.MapReduce.OutputToCollection;
+using Raven.Server.Documents.Indexes.MapReduce.Static;
+using Raven.Server.Exceptions;
+using Raven.Server.Utils;
+using Sparrow.Json;
+using Sparrow.Logging;
+using Sparrow.Server.Exceptions;
+using Voron.Impl;
+namespace Raven.Server.Documents.Indexes.Persistence.Corax;
+
+public class OutputReduceCoraxIndexWriteOperation : CoraxIndexWriteOperation
+{
+    private readonly OutputReduceToCollectionCommandBatcher _outputReduceToCollectionCommandBatcher;
+    private readonly TransactionHolder _txHolder;
+
+    public OutputReduceCoraxIndexWriteOperation(MapReduceIndex index, Transaction writeTransaction, CoraxDocumentConverterBase converter, Logger logger, JsonOperationContext indexContext) : base(index,
```

Comment:
Let's talk directly about trying to avoid the code duplication here

Reply:
done